tests/unit/test_hardware.py

```diff
@@ -0,0 +1,61 @@
+"""
+..
+   test_hardware.py
+
+Unit tests for pyra.hardware.py.
+"""
+# local imports
+from pyra import hardware
+
+
+def test_update_cpu():
+    """Tests if update_cpu returns a float value"""
+    test_cpu_percent = hardware.update_cpu()
+    assert test_cpu_percent <= 100  # test if value is number with a max value of 100
+    assert test_cpu_percent >= 0  # test if value is number with a min value of 0
+
+
+def test_update_gpu():
+    """
+    # todo... this test needs to be defined
+    """
+    pass
+
+
+def test_update_memory():
+    """Tests if update_memory returns a float value"""
+    test_memory_percent = hardware.update_memory()
```

Comment:
Just an idea, you could try allocating a big thing in memory and seeing if the memory usage goes up.